templates/question.html

```diff
@@ -22,8 +22,9 @@
       <p>{{ ngettext('This <strong>must be corrected</strong> to continue', 'These <strong>must be corrected</strong> to continue', form.mapped_errors | length) }}.</p>
       <ul class="list list--bare">
         {% for answer_id, error in form.mapped_errors %}
+          {% set hrefValue = answer_id + '-error' if answer_id != 'date-range-question' else answer_id %}
```

Comment:
Using a fixed answer id of `date-range-question` won't work as the answer id can be anything a survey author wants it to be. I'm not sure it matters though as it looks like the error panel for date ranges have an id of `<answer_id>-error` anyway?

Reply:
OK. In this scenario the `-error` suffix isn't required....

Second date fieldset contains a date ahead of the first fieldset.

No error panel is rendered around the inputs so the `-error` id isn't generated for the panel. 

The href for the error link uses the question id and not the answer id (already doing that and only for this scenario) so that it jumps to the question block and the js finds the input to focus.

If there's some other value we could test for instead of `date-range-question` then let me know and I will update.

The main thing is that we don't want the `-error` on the href for the date range scenario.

Reply:
Ah, I hadn't noticed that. I've pushed a change to move the error id generation into the code, where we already know about the distinction between question and answer errors.